File: apps/backend/Models/Account.cs
using System.ComponentModel.DataAnnotations;
using DigitalLedger.Api.Models.Enums;

namespace DigitalLedger.Api.Models;

public class Account
{
    public Guid Id { get; set; }

    [Required, MaxLength(20)]
    public required string Code { get; set; }

    [Required, MaxLength(100)]
    public required string Name { get; set; }

    [Required]
    public AccountCategory Category { get; set; }

    [Required]
    public NormalBalance NormalBalance { get; set; }

    public string? Description { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: apps/frontend/src/types/account.ts
export type AccountCategory = 'ASSET' | 'LIABILITY' | 'EQUITY' | 'REVENUE' | 'EXPENSE';
export type NormalBalance = 'DEBIT' | 'CREDIT';

export interface Account {
  id: string;
  code: string;
  name: string;
  category: AccountCategory;
  normalBalance: NormalBalance;
  description?: string;
  isActive: boolean;
  createdAt: string;
}

export interface CreateAccountDto {
  code: string;
  name: string;
  category: AccountCategory;
  description?: string;
}

export interface UpdateAccountDto {
  name: string;
  description?: string;
}
